phase1-coordinator/src/storage/disk.rs

```diff
@@ -146,13 +122,7 @@ impl Disk {
         let mut manifest = self.manifest.write().unwrap();
 
         // Create the new file.
-        let file = manifest.create_file(&locator, Some(size))?;
-
-        // Add the file to the locators.
-        self.open.insert(
-            locator.clone(),
-            Arc::new(RwLock::new(unsafe { MmapOptions::new().map_mut(&file)? })),
-        );
+        let _ = manifest.create_file(&locator, Some(size))?;
```

Comment:
I think that would make Disk a leaky abstraction. To address this I guess we could modify the abstraction instead to rely on Reader Writer traits instead of byte array read/write?